cocos/2d/CCAction.cpp

```diff
@@ -123,21 +128,29 @@ bool Speed::initWithAction(ActionInterval *action, float speed)
 Speed *Speed::clone() const
 {
     // no copy constructor
-    auto a = new (std::nothrow) Speed();
-    a->initWithAction(_innerAction->clone(), _speed);
-    a->autorelease();
-    return a;
+    if (_innerAction) {
+        return Speed::create(_innerAction->clone(), _speed);
+    }
+    
+    return nullptr;
```

Comment:
It is strange to return null for clone function. And because initWithAction() will return false when inner action is null, so the `_innerAction` can not be a null.